src/transport/SecureSessionMgr.h

```diff
@@ -297,4 +297,49 @@ class DLL_EXPORT SecureSessionMgr : public TransportMgrDelegate
     static void ExpiryTimerCallback(System::Layer * layer, void * param, System::Error error);
 };
 
+namespace MessagePacketBuffer {
+/**
+ * Maximum size of a message footer, in bytes.
+ */
+constexpr uint16_t kMaxFooterSize = kMaxTagLen;
+
+/**
+ * Allocates a packet buffer with space for message headers and footers.
+ *
+ *  Fails and returns \c nullptr if no memory is available, or if the size requested is too large.
+ *
+ *  @param[in]  aAvailableSize  Minimim number of octets to for application data.
+ *
+ *  @return     On success, a PacketBufferHandle to the allocated buffer. On fail, \c nullptr.
+ */
+inline System::PacketBufferHandle New(size_t aAvailableSize)
+{
+    return System::PacketBufferHandle::New(aAvailableSize + kMaxFooterSize);
```

Comment:
Should guard against overflow on the addition here...

Reply:
Right. Eventually I might learn to check.